File: BluetoothLightLib/src/main/java/com/telink/bluetooth/event/ServiceEvent.java
/*
 * Copyright (C) 2015 The Telink Bluetooth Light Project
 *
 */
package com.telink.bluetooth.event;

import android.os.IBinder;

/**
 * LightService事件
 */
public class ServiceEvent extends DataEvent<IBinder> {

    /**
     * 服务启动
     */
    public static final String SERVICE_CONNECTED = "com.telink.bluetooth.light.EVENT_SERVICE_CONNECTED";
    /**
     * 服务关闭
     */
    public static final String SERVICE_DISCONNECTED = "com.telink.bluetooth.light.EVENT_SERVICE_DISCONNECTED";

    public ServiceEvent(Object sender, String type, IBinder args) {
        super(sender, type, args);
    }

    public static ServiceEvent newInstance(Object sender, String type, IBinder args) {
        return new ServiceEvent(sender, type, args);
    }
}

File: BluetoothLightLib/src/main/java/com/telink/bluetooth/event/ServiceEvent.kt
package com.telink.bluetooth.event

import android.os.IBinder

/**
 * LightService事件
 */
class ServiceEvent(sender: Any, type: String, args: IBinder?) : DataEvent<IBinder>(sender, type, args) {
    companion object {

        //服务启动
        val SERVICE_CONNECTED = "event_service_connected"
        //服务关闭
        val SERVICE_DISCONNECTED = "event_service_disconnected"

        fun newInstance(sender: Any, type: String, args: IBinder?): ServiceEvent {
            return ServiceEvent(sender, type, args)
        }
    }
}